src/journal/journalctl.c

```diff
@@ -1804,6 +1820,22 @@ static int add_syslog_identifier(sd_journal *j) {
         return 0;
 }
 
+static int add_exclude_identifier(sd_journal *j) {
+        int r;
+
+        assert(j);
+
+        j->exclude = set_free(j->exclude);
+
+        STRV_FOREACH(i, arg_exclude_identifier) {
+                r = set_ensure_put(&j->exclude, &string_hash_ops, *i);
```

Comment:
I noticed that `j->exclude` references strings owned by `arg_exclude_identifier`.

As you can see, `arg_exclude_identifier` is freed in main() *before* `sd_journal *j` is freed.

Please use `set_put_strdup()`.

Reply:
It's meaningless:
1. `set_put_strdup()` (here more applicable `set_put_strdup_full()` as we want specify hash_ops for strings is:
```
130 int _set_put_strdup_full(Set **s, const struct hash_ops *hash_ops, const char *p  HASHMAP_DEBUG_PARAMS);
131 #define set_put_strdup_full(s, hash_ops, p) _set_put_strdup_full(s, hash_ops, p  HASHMAP_DEBUG_SRC_ARGS)
132 #define set_put_strdup(s, p) set_put_strdup_full(s, &string_hash_ops_free, p)
```
2. `_set_put_strdup_full()` is:
```
1844 int _set_put_strdup_full(Set **s, const struct hash_ops *hash_ops, const char *p  HASHMAP_DEBUG_PARAMS) {
...
1851         r = _set_ensure_allocated(s, hash_ops  HASHMAP_DEBUG_PASS_ARGS);
...
1858         c = strdup(p);
...
1862         return set_consume(*s, c);
```
3. `set_consume()` is:
```
1797 int set_consume(Set *s, void *value) {
1798         int r;
1799
1800         assert(s);
1801         assert(value);
1802
1803         r = set_put(s, value);
1804         if (r <= 0)
1805                 free(value);
1806
1807         return r;
1808 }
```
So it calls `free()` of a value after calling `set_put()`

In this PR `free()` is called by `strv_free()`, while `set_put()` is called directly via `set_ensure_put()`

Reply:
I'll send a commit soon